Tools/LocomotionTool/LocomotionTool.cs

```diff
@@ -0,0 +1,559 @@
+
+								var rayToRay = otherPosition - thisPosition;
+								var midPoint = thisPosition + rayToRay * 0.5f;
+
+								rayToRay.y = 0; // Use for yaw rotation
+
+								var pivotYaw = MathUtilsExt.ConstrainYawRotation(cameraRig.rotation);
+
+								if (!m_Scaling)
+								{
+									m_StartScale = this.GetViewerScale();
+									m_StartDistance = distance;
+									m_StartMidPoint = pivotYaw * midPoint * m_StartScale;
+									m_StartPosition = cameraRig.position;
+									m_StartDirection = rayToRay;
+									m_StartYaw = cameraRig.rotation.eulerAngles.y;
+
+									locomotionTool.m_Scaling = true;
+
+									CreateViewerScaleVisuals(rayOrigin, otherRayOrigin);
+								}
+
+								m_Scaling = true;
+
+								var currentScale = Mathf.Clamp(m_StartScale * (m_StartDistance / distance), k_MinScale, k_MaxScale);
+
+								var thumb = m_LocomotionInput.thumb;
+								var thumbHeld = thumb.isHeld;
+								if (thumbHeld)
+									consumeControl(thumb);
+
+								var otherThumb = otherLocomotionInput.thumb;
+								var otherThumbHeld = otherThumb.isHeld;
+								if (otherThumbHeld)
+									consumeControl(otherThumb);
+
+								// Press both thumb buttons to reset scale
+								if (thumbHeld && otherThumbHeld)
+								{
+									m_AllowScaling = false;
+
+									rayToRay = otherRayOrigin.position - rayOrigin.position;
+									midPoint = rayOrigin.position + rayToRay * 0.5f;
+									var currOffset = midPoint - cameraRig.position;
+
+									cameraRig.position = midPoint - currOffset / currentScale;
+									cameraRig.rotation = Quaternion.AngleAxis(m_StartYaw, Vector3.up);
+
+									ResetViewerScale();
+								}
+
+								var trigger = m_LocomotionInput.trigger;
+								var triggerHeld = trigger.isHeld;
+								if (triggerHeld)
+									consumeControl(trigger);
+
+								var otherTrigger = otherLocomotionInput.trigger;
+								var otherTriggerHeld = otherTrigger.isHeld;
+								if (otherTriggerHeld)
+									consumeControl(otherTrigger);
+
+								// Press both triggers to reset to origin
+								if (triggerHeld && otherTriggerHeld)
+								{
+									m_AllowScaling = false;
+
+									cameraRig.position = Vector3.up * VRView.HeadHeight;
+									cameraRig.rotation = Quaternion.identity;
+
+									ResetViewerScale();
+								}
+
+								if (m_AllowScaling)
+								{
+									var yawSign = Mathf.Sign(Vector3.Dot(Quaternion.AngleAxis(90, Vector3.down) * m_StartDirection, rayToRay));
+									var currentYaw = m_StartYaw + Vector3.Angle(m_StartDirection, rayToRay) * yawSign;
+									var currentRotation = Quaternion.AngleAxis(currentYaw, Vector3.up);
+									midPoint = currentRotation * midPoint * currentScale;
+
+									cameraRig.position = m_StartPosition + m_StartMidPoint - midPoint;
+									cameraRig.localScale = Vector3.one * currentScale;
+									cameraRig.rotation = currentRotation;
+
+									m_MainCamera.nearClipPlane = m_OriginalNearClipPlane * currentScale;
```

Comment:
Hm. I definitely saw issues with this when scaling myself down very small. Let me test on my end